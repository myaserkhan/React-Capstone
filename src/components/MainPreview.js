import React, { useEffect, useState } from 'react';
import { NavLink } from 'react-router-dom';
import axios from 'axios';
import './stylesheets/MainPreview.scss';
import pokeball from '../assets/pokeball.png';

function MainPreview() {
  const [img, setImg] = useState(pokeball);
  const [ID, setID] = useState();
  const randomID = () => Math.floor(Math.random() * 897 + 1);
  const id = randomID();

  useEffect(() => {
    async function fetchData() {
      const response = await axios.get(
        `https://pokeapi.co/api/v2/pokemon/${id}`,
      );
      setID(response.data.id);
      setImg(response.data.sprites.other['official-artwork'].front_default);
    }
    fetchData();
  }, []);

  const random = () => {
    const random = Math.floor(Math.random() * (55 - 43) + 43);
    return { backgroundColor: `hsl(219,45%,${random}%)`, textDecoration: 'none' };
  };

  return (
    <NavLink to={`/pokemon/details/${ID}`} className="guessBox" style={random}>
      <div className="guessOval">
        <h1>{ID < 10 ? `0${ID}` : ID}</h1>
        <img src={img} alt="" />
      </div>
      <h2 className="subtitle">Who&apos;s this POKEMON?</h2>
    </NavLink>
  );
}

export default MainPreview;
