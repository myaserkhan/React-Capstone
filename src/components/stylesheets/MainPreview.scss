@import '../../App.scss';

.container {
  text-align: center;
}

.guessBox {
  display: flex;
  flex-direction: row;
  justify-content: center;
  align-items: center;
  height: 215px;
  border: 1px solid grey;
  color: $clr-white;
  position: relative;
  cursor: pointer;

  .guessOval {
    background-color: $clr-white;
    border-radius: 50%;
    padding: 1rem;

    h1 {
      font-size: 1.25rem;
      position: absolute;
      top: 0;
      left: 0;
      color: $clr-white;
      border: 1px solid $clr-white;
      border-radius: 50%;
      padding: 0.25rem;
    }

    .hide {
      width: 100px;
      filter: grayscale(10%);
      filter: brightness(0%);
    }

    .reveal {
      width: 100px;
    }
  }

  .subtitle {
    color: $clr-yellow-gold;
    font-size: 2.5rem;
    text-decoration: none;
    text-align: center;
  }
}

@media screen and (min-width: 768px) {
  .guessBox {
    padding: 0 4rem;

    .guessOval {
      padding: 2rem;
    }

    .subtitle {
      margin-left: 1rem;
      font-size: 2.5rem;
      text-decoration: none;
      text-align: center;
    }
  }
}
