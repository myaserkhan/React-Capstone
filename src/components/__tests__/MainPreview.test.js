import { render, screen } from '@testing-library/react';
import renderer from 'react-test-renderer';
import { Provider } from 'react-redux';
import { BrowserRouter } from 'react-router-dom';
import MainPreview from '../MainPreview';
import store from '../redux/configureStore';

const MockMainPreview = () => (
  <Provider store={store}>
    <BrowserRouter>
      <MainPreview pokemon={{ id: '1', name: 'bulbasaur' }} />
    </BrowserRouter>
  </Provider>
);

describe('Test the MainPreview component', () => {
  test('renders correctly', () => {
    const tree = renderer.create(<MockMainPreview />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  test('renders heading', () => {
    render(<MockMainPreview />);
    const headingElement = screen.getByRole('heading', {
      name: /who's that pokemon\?/i,
    });
    expect(headingElement).toBeInTheDocument();
  });
});
