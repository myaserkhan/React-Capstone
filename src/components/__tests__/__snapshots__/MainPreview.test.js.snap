// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test the MainPreview component renders correctly 1`] = `
<section
  className="container"
>
  <a
    className="guessBox"
    href="/pokemon/details/undefined"
    onClick={[Function]}
    style={
      Object {
        "textDecoration": "none",
      }
    }
  >
    <div
      className="guessOval"
    >
      <h1 />
      <img
        alt=""
        className="hide"
        src="pokeball.png"
      />
    </div>
    <h2
      className="subtitle"
    >
      Who's that POKEMON?
    </h2>
  </a>
  <form
    onSubmit={[Function]}
  >
    <input
      onChange={[Function]}
      placeholder="Type your Guess here"
      type="text"
    />
    <button
      type="submit"
    >
      Check Answer
    </button>
  </form>
</section>
`;
